app/generator/api_combiner.rb

```diff
@@ -1,27 +1,57 @@
 module ApiCombiner
+  attr_reader :trail_species
+
   def self.get_species(trail_id)
-    gbif = GbifService.new
-    eol = EolService.new
     trail = Trail.find(trail_id)
-    trail_species = []
+    @trail_species = []
+
+    add_gbif_species(trail.id, trail.lat, trail.lng)
+    add_xeno_canto_species(trail.id, trail.lat, trail.lng)
+  end
 
-    gbif_species = gbif.species(trail.lat, trail.lng)
+  def self.add_gbif_species(trail_id, lat, lng)
+    xeno_canto = XenoCantoService.new
+    gbif = GbifService.new
+    eol = EolService.new
+    gbif_species = gbif.species(lat, lng)
     gbif_species.each do |species|
       if species["species"]
         genus, spec = species["species"].split(" ")
         img = eol.image_from_species(genus, spec)
 
-        trail_species << Species.create(
+        @trail_species << Species.create(
           trail_id: trail_id,
           kingdom: species["kingdom"],
           scientific_name: species["species"],
           common_name: species["vernacularName"],
           photo_url: img,
-          clip_url: "not available",
+          clip_url: xeno_canto.species_recording(species["species"]),
           lat: species["decimalLatitude"],
           lng: species["decimalLongitude"])
       end
     end
-    trail_species
+  end
+
+  def self.add_xeno_canto_species(trail_id, lat, lng)
```

Comment:
Assignment Branch Condition size for add_xeno_canto_species is too high. [20.42/15]